vulnerabilities/importers/npm.py

```diff
@@ -9,190 +9,140 @@
-        files = self._updated_files.union(self._added_files)
-        advisories = []
-        for f in files:
-            processed_data = self.process_file(f)
-            if processed_data:
-                advisories.extend(processed_data)
-        return self.batch_advisories(advisories)
-
-    def set_api(self, packages):
-        asyncio.run(self._versions.load_api(packages))
-
-    def collect_packages(self):
-        packages = set()
-        files = self._updated_files.union(self._added_files)
-        for f in files:
-            data = load_json(f)
-            packages.add(data["module_name"].strip())
+        advisory_reference = Reference(
+            url=f"https://github.com/nodejs/security-wg/blob/main/vuln/npm/{id}.json",
+            reference_id=id,
+            severities=severities,
+        )
 
-        return packages
+        for ref in data.get("references") or []:
+            references.append(
+                Reference(
+                    url=ref,
+                    severities=severities,
+                )
+            )
 
-    @property
-    def versions(self):  # quick hack to make it patchable
-        return self._versions
+        if advisory_reference not in references:
+            references.append(advisory_reference)
+
+        package_name = data.get("module_name")
+        affected_packages = []
+        if package_name:
+            affected_packages.append(self.get_affected_package(data, package_name))
+        advsisory_aliases = data.get("cves") or []
+
+        for alias in advsisory_aliases:
+            yield AdvisoryData(
+                summary=build_description(summary=summary, description=description),
+                references=references,
+                date_published=date_published,
+                affected_packages=affected_packages,
+                aliases=[alias],
+            )
 
-    def process_file(self, file) -> List[AdvisoryData]:
+    def get_affected_package(self, data, package_name):
+        vulnerable_range = data.get("vulnerable_versions") or ""
 
-        record = load_json(file)
-        advisories = []
-        package_name = record["module_name"].strip()
+        # https://github.com/nodejs/security-wg/blob/main/vuln/npm/213.json#L14
+        if vulnerable_range == "<=99.999.99999":
+            vulnerable_range = "*"
 
-        publish_date = parse(record["updated_at"])
-        publish_date = publish_date.replace(tzinfo=pytz.UTC)
+        affected_version_range = (
+            NpmVersionRange.from_native(vulnerable_range) if vulnerable_range else None
+        )
 
-        all_versions = self.versions.get(package_name, until=publish_date).valid_versions
-        aff_range = record.get("vulnerable_versions")
-        if not aff_range:
-            aff_range = ""
-        fixed_range = record.get("patched_versions")
-        if not fixed_range:
-            fixed_range = ""
+        if vulnerable_range == "*":
```

Comment:
Why singling this star out ? why not folding this  with the code below... fixed_version is None always if range is "*"